### PR TITLE
[Editor] Providing front end design for Siddhi extension-installation features.

### PR DESCRIPTION
**Pull request by SinthujanSintha**
## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

innerModalBox
search bar
alertBox

### CODE REVIEW

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/dialog/extension-install-dialog.js [hunk 1]
```diff
@@ -2,39 +2,34 @@
...
+        var ExtensionInstallDialog = function ()
+             {
+            /**
+             * @constructs
+             * @class ExtensionInstallDialogExtensionInstallDialog
+             * @param {Object} config configuration options for ExtensionInstallDialog
+             */
+            this.initialize = function (options) {
+                this.dialog_containers = $(_.get(options.config.dialog, 'container'));
+                this.extensionList = getExtensionDetail();
```
Comment (dnwick) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/dialog/extension-install-dialog.js]: change method name to getExtensionDetails

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/dialog/extension-install-dialog.js [hunk 2]
```diff
@@ -363,8 +256,52 @@ define(['require', 'lodash', 'jquery', 'log', 'backbone'],
                         self._extensionPartialModel.modal('show');
                     }
 
-                },
-            });
+                }
+
+        };
+
+        var constants = {
```
Comment (dnwick) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/dialog/extension-install-dialog.js]: move the constants to top or in the initialize method

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/dialog/extension-install-dialog.js [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+                                "<button type='button' class='close' data-dismiss='modal' aria-label='Close'>" +
+                                "<i class=\"fw fw-cancel  about-dialog-close\"></i>" +
+                                "</button>" +
+                                '<h3 class="modal-title file-dialog-title" id="partialExtenName">'
+                                + extension.name +
+                                '</h3> ' +
+                                '<hr class="style1"> ' +
+                                '</div>' +
+                                ' <div class="modal-body">' +
+                                ' <h3>Description</h3>' +
```
Comment (dnwick) [inline tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/dialog/extension-install-dialog.js]: Try to add a long description and see how UI behaves ,, whether they aligned correctly if not fix them using css

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/dialog/extension-install-dialog.js (hunk 1 above)
Comment (SinthujanSintha) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/dialog/extension-install-dialog.js]: 9ed07404e8a403916fc57b790dd8f95a62784477 this commit  will be resolving above request.

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/dialog/extension-install-dialog.js (hunk 2 above)
Comment (SinthujanSintha) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/dialog/extension-install-dialog.js]: 9ed0740 this commit will be resolving above request.

File: tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/dialog/extension-install-dialog.js (hunk 3 above)
Comment (SinthujanSintha) [reply tooling/components/io.siddhi.distribution.editor.core/src/main/resources/web/editor/commons/js/dialog/extension-install-dialog.js]: 9ed0740 this commit will be resolving above request.
InfoModel

**Pull request merged**